eth/protocols/eth/peer.go

```diff
@@ -369,6 +400,39 @@ func (p *Peer) AsyncSendNewBlock(block *types.Block, td *big.Int) {
 	}
 }
 
+// SendVotes propagates a batch of votes to the remote peer.
+func (p *Peer) SendVotes(votes []*types.VoteEnvelope) error {
+	// Mark all the transactions as known, but ensure we don't overflow our limits
+	for p.knownVotes.Cardinality() > max(0, maxKnownTxs-len(votes)) {
+		p.knownVotes.Pop()
+	}
+	for _, vote := range votes {
+		p.knownVotes.Add(vote.Hash())
+	}
+	return p2p.Send(p.rw, VotesMsg, &VotesPacket{votes})
+}
+
+// AsyncSendVotes queues a batch of vote hashes for propagation to a remote peer. If
+// the peer's broadcast queue is full, the event is silently dropped.
+func (p *Peer) AsyncSendVotes(votes []*types.VoteEnvelope) {
+	select {
+	case p.voteBroadcast <- votes:
+		// Mark all the transactions as known, but ensure we don't overflow our limits
+		for p.knownVotes.Cardinality() > max(0, maxKnownVotes-len(votes)) {
```

Comment:
Suggest to check if maxKnownVotes > len(votes), otherwise maxKnownVotes-len(votes) will be a very large number due to uint64 overflow

Reply:
its int here. And we need the max number not just return true or false